src/eos-autoupdater.c

```diff
@@ -212,7 +213,8 @@ update_stamp_file (guint update_interval_days,
     return;
   }
 
-  g_get_current_time (&mtime);
+  mtime.tv_sec = last_successful_update_secs;
```

Comment:
Hm, we should add an `else` case to the `if (randomized_delay > 0)` that sets the mtime of the stamp file to `last_successful_update_secs`, otherwise the mtime of the stamp will be now when there is no randomized delay, instead of `last_successful_update_secs`, no? Earlier it wasn't a problem, I think, because g_get_current_time() was called around the same time when the contents of the timestamp were updated.